AutoGear.lua

```diff
@@ -3935,6 +3945,7 @@ function AutoGearGetBest1hPairing(info)
 end
 
 function AutoGearTooltipHook(tooltip)
+	if tooltip ~= GameTooltip or tooltip ~= ShoppingTooltip1 or tooltip ~= ShoppingTooltip2 or tooltip ~= ItemRefTooltip or tooltip == nil then return end
```

Comment:
This logic does not make sense.  If the tooltip is `ShoppingTooltip1`, the first condition, `tooltip ~= GameTooltip` will immediately evaluate as true, resulting in executing `return`.  You need `and` there.

This makes me concerned that this change wasn't tested.  The shopping tooltips are the comparison tooltips when hovering gear that is not equipped.

By the way, are you sure we can't instead simply check for whether the tooltip data type is `Enum.TooltipDataType.Item`?  I imagine that if they added that enum, they would have also added a way to check it.

Come to think of it, the hook itself  (this line: `TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, AutoGearTooltipHook)`) seems to exclude tooltips which are not of tooltip data type `Enum.TooltipDataType.Item`, so what makes this additional check necessary at all?  It did not exist in the original code, since the original code only hooked the tooltips it needed to, and the new code seems to also do that.